src/api/users/mutations/loginUser/index.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+import bcrypt from 'bcrypt'
+import type { ValidationError } from '@types/ValidationError'
+import createToken from '@models/Token/createToken'
+import retrieveToken from '@models/Token/retrieveToken'
+import User from '@models/User'
+import { AUTHENTICATION_ERROR_INVALID_CREDENTIALS } from '@constants/errors'
+
+export type LoginUserArgs = {
+  email: string,
+  password: string
+}
+
+type LoginUserResponse = {
+  token?: string,
+  errors: Array<ValidationError>
+}
+
+const resolver = async (
+  _: any,
+  args: LoginUserArgs
+): Promise<LoginUserResponse> => {
+  const errors: Array<ValidationError> = []
+  const email: string = args.email
+  const password: string = args.password
+
+  if (!email) {
+    errors.push({ field: 'email', message: 'This field must not be empty' })
+  }
+
+  if (!password) {
+    errors.push({ field: 'password', message: 'This field must not be empty' })
+  }
```

Comment:
We don’t need those two validations. The schema enforces it for us and FE will provide client-side validation.

Reply:
```
mutation {
  loginUser(email:"", password:"") {errors{message}}
}
```
Will not throw the proper error (which is that the field is empty) if we remove validation (it will throw incorrect credentials)
Also, client-side validation is not validation

Reply:
Didn’t think about empty strings 🙀